graph/bellman-ford.py

```diff
@@ -0,0 +1,149 @@
+class OrientedGraph:
+	def __init__(self):
+		"""
+		Init Oriented graph object
+		"""
+		self.nodes = {}
+
+	def insertNode(self, u):
+		"""
+		Insert node u
+
+		Parameters:
+		u : str
+		"""
+		if u not in self.nodes:
+			self.nodes[u] = {}
+
+	def V(self):
+		"""
+		Return veritces
+
+		Return:
+		nodes : dict
+		"""
+		return self.nodes.keys()
+
+	def adj(self, u):
+		if u in self.nodes:
+			return self.nodes[u]
+
+	def insertEdge(self, u, v, w=0):
+		"""
+		Insert edge from u to v
+
+		Parameters:
+		u : str
+		   Start node
+		v: str
+		   End node
+		w: float
+		   Weight
+		"""
+		if u not in self.nodes:
+			self.insertNode(u)
+		if v not in self.nodes:
+			self.insertNode(v)
+		self.nodes[u][v] = w
+
+	def bfs(self, start):
+		"""
+		Perform BFS search
+
+		Parameters:
+		start: str
+		"""
+		S = [start]
+		visited = {}
+		for node in self.V():
+			visited[node] = False
+		visited[start] = True
+		while len(S) > 0:
+			u = S.pop(0)
```

Comment:
Could you consider using `collections.deque` here? since poping from the left on a regular python list incurs in shifting the entire list to the left so its O(n)